pkg/provider/aws/resources/ecr.go

```diff
@@ -37,7 +37,7 @@ type RepoCreateParams struct {
 
 func (repo *EcrRepository) Create(dag *core.ResourceGraph, params RepoCreateParams) error {
 	repo.Name = params.AppName
-	repo.ConstructsRef = params.Refs
+	repo.ConstructsRef = params.Refs.Clone()
```

Comment:
We should decide whether cloning happens in `Create` or caller-side. There's reasonable arguments for either, so we should pick one and standardize.

Reply:
Not necessarily memory allocation, but just for standardization and consistency - even if that standard is "clone on both sides to be sure".